bc_obps/service/operation_service_v2.py

```diff
@@ -415,11 +415,15 @@ def check_conditions() -> Generator[Tuple[Callable[[], bool], str], None, None]:
     @classmethod
     def generate_boro_id(cls, user_guid: UUID, operation_id: UUID) -> Optional[BcObpsRegulatedOperation]:
         operation = OperationService.get_if_authorized(user_guid, operation_id)
-        is_eio = operation.registration_purpose == Operation.Purposes.ELECTRICITY_IMPORT_OPERATION
+        is_regulated_operation = operation.registration_purpose in [
+            Operation.Purposes.OBPS_REGULATED_OPERATION,
+            Operation.Purposes.NEW_ENTRANT_OPERATION,
+            Operation.Purposes.OPTED_IN_OPERATION,
+        ]
```

Comment:
I believe we may need this check in other areas in the future, so it would be great to move it to a function under the Operation data model(or a util file for the Operation data model). Additionally, it would be helpful to implement a constraint in the data model to prevent generating Boro IDs for non-eligible purposes.

Reply:
I made it a property on the `operation` model and moved the checks that were previously in the endpoint into the property. The status check is still in the endpoint because reg 1 had different rules

Reply:
Good catch!

Reply:
Actually, it's the same situation for checking if an operation is regulated. Reg1 doesn't use registration purposes, so if I put the check on the db, I'll break reg1